TFGCasper.thy

```diff
@@ -0,0 +1,79 @@
+theory TFGCasper
+
+imports Main HOL.Real CBCCasper ConsensusSafety ExampleProtocols
+
+begin
+
+(* Section 4.4: Casper the Friendly GHOST *)
+(* Definition 4.23: Blocks *)
+type_synonym block = consensus_value
+
+locale Ghost = Protocol +
+  fixes Genesis :: block
+  and NextBlkgen :: "block \<Rightarrow> block set"
```

Comment:
Variables should be `snake_case` so `genesis` and `next_blk_generator` are better?

Also, we can omit `next` and jsut say `block_generator`.

Reply:
`genesis` and `block_generator` look better for me 👍

Reply:
ea91df7330b1c871e5312d398e53160033b77b44